app/main/controllers/employee.py

```diff
@@ -0,0 +1,43 @@
+from flask import request
+from flask_restx import Resource
+
+from ..utils.dto import EmployeeDto
+from ..services.employee_service import (get_employee_by_id,
+                                         get_all_employees)
+
+from ..models.employee import Employee as EmployeeModel
+
+api = EmployeeDto.api
+_employee = EmployeeDto.employee
+
+# TODO: add argument parsing functionality. Note that Flasks's reqparse is depricated,
+# so should use marshmallow or similar instead.
+
+@api.route("",
+           doc={
+               "description": "Retreive a list of employees.",
+           })
+class Employees(Resource):
+    def get(self):
+        return list(get_all_employees())
+
+    def post(self):
```

Comment:
Lets add more swagger decorators:

```python
@api.doc('Add employee')
@api.expect(_new_employee, validate=True)
@api.marshal_with(_employee)
```
You'll need to create a `_new_employee` api model in the `dto.py`. This is intended to define the post data requirements. @Collinbrown95, you did mention something about request parsing. So I'm not sure how that fits with this, but I still feel the endpoint documentation should specify what it expects.

Reply:
Request parsing is used to validate the data in the request body before it reaches a particular endpoint. If there is an error in the request body (e.g. a field is missing), then it automatically catches it and sends a response back to the user with a message on how to resolve it.

That being said, as long as the request parsing is consistent with the model description, I don't think there should be an issue. I will test this and confirm.

Reply:
__Correction:__ after looking at the [flask-restx docs](https://flask-restx.readthedocs.io/en/latest/swagger.html#documenting-with-the-api-doc-decorator), it actually looks like using the ```@api.expect``` decorator and using a request parser are equivalent.

Given that we are already defining the api models in ```dto.py```, my thought is that the ```@api.expect``` approach would avoid some duplication of work. I'll play around with this feature more over the next while and see if there are any shortcomings that I'm not aware of.